File: binary_search.py
def b_search(nums, target, l=None, r=None):
    if l is None:
        l, r= 0, len(nums) - 1
    m = (l+r)//2
    if l <= r:
        if nums[m] >  target:
            return b_search(nums, target, l=l, r=m-1)
        elif nums[m] < target:
            return b_search(nums, target, l=m+1, r=r)
        elif nums[m] == target:
            return m
    return -1

x = [n for n in range(1, 10000000)]
print(b_search(x, 700648))

File: C++/binary_search.cpp
#include <iostream>
#include <vector>

using namespace std;

int b_search(vector<int> arr, int target){
    int l = 0, r = arr.size() - 1;
    while (l <= r){
        int mid = (l + r) / 2;
        if (arr[mid] < target){
            l = mid + 1;
        }
        else if (arr[mid] > target){
            r = mid - 1;
        }
        else if (arr[mid] == target){
            return mid;
        }
    }
    return -1;
}

int main(){
    vector<int> x = {1,2,4,6,22,36,42,77};
    int target;
    cout<<"Enter a number to search in the array: ";
    cin>>target;
    cout<<b_search(x, target)<<endl;
    return 0;
}